scripts/plugins/test_extauth_hook_AD.py

```diff
@@ -103,7 +103,27 @@ def test_permit_admin_user(self, mock_rename, mock_chmod):
         user = build_user("CONNAPP", "radmin", True)
         mock_session.xenapi.subject.get_record.return_value = user
         permit_user = r"account sufficient pam_succeed_if.so user = CONNAPP\radmin"
-        dynamic = DynamicPam(mock_session, args_bd_wibind)
+        dynamic = DynamicPam(mock_session, args_bd_winbind)
```

Comment:
You need a test case with the name containing a plus, if that is legal.

Reply:
I suspect that names containing a plus sign are not valid, should xapi reject them?

Reply:
yes, plus is not permitted in windows AD, this means it will never happens in username, (if user force it, then AD will replace plus with "_").
xapi can detect and reject it. But that may not be necessary as
1. the username are queried from AD, not input from user, in this case, plus never exists
2. In case of plus do exists, we just put it into the pam configure file, and it does not affect other users. This is similar to detect and skip this `bad` user.